pkg/queue/cluster_queue_impl.go

```diff
@@ -221,6 +221,21 @@ func (c *clusterQueueBase) DumpInadmissible() (sets.Set[string], bool) {
 	return elements, true
 }
 
+func (c *clusterQueueBase) Snapshot(maxCount int32) ([]*kueue.Workload, bool) {
```

Comment:
I would suggest to represent the snapshot as a dedicated struct, say "pendingWorkloadsSnapshot".